front/lib/swr/conversations.ts

```diff
@@ -567,3 +569,110 @@ export function useConversationMarkAsRead({
     markAsRead,
   };
 }
+
+type ConversationAction = "join" | "leave" | "delete";
+
+export const useConversationMenu = ({
+  ownerId,
+  conversationId,
+  userId,
+}: {
+  ownerId: string;
+  conversationId: string | null;
+  userId: string | null;
+}): {
+  joinConversation: () => Promise<boolean>;
+  action: ConversationAction;
+  users: UserParticipantType[];
+  agents: AgentParticipantType[];
+} => {
+  const sendNotification = useSendNotification();
+
+  const { mutateConversations } = useConversations({
+    workspaceId: ownerId,
+  });
+  const { mutateConversationParticipants, conversationParticipants } =
+    useConversationParticipants({
+      conversationId,
+      workspaceId: ownerId,
+      options: { disabled: conversationId === null },
+    });
+
+  const isUserParticipating =
+    userId !== null &&
+    conversationParticipants?.users.find(
+      (participant) => participant.sId === userId
+    );
+
+  const isLastParticipant =
+    isUserParticipating && conversationParticipants?.users.length === 1;
+
+  const joinConversation = useCallback(async (): Promise<boolean> => {
+    if (!conversationId) {
+      return false;
+    }
+    try {
```

Comment:
can we have the try catch only around the potentially-failing functions ? (i.e fetch)

see CODING_RULES.md [ERR1]

Reply:
so early return on the catch and then treat the happy-scenario outside of try